packages/next-international/__tests__/use-scoped-i18n.test.tsx

```diff
@@ -0,0 +1,245 @@
+import { createI18n } from '../src';
+import { render, screen } from './utils';
+import en from './utils/en';
+
+beforeEach(() => {
+  vi.mock('next/router', () => ({
+    useRouter: vi
+      .fn()
+      .mockImplementationOnce(() => ({
+        locales: ['en', 'fr'],
+      }))
+      .mockImplementationOnce(() => ({
+        locale: 'en',
+        defaultLocale: 'en',
+      }))
+      .mockImplementation(() => ({
+        locale: 'en',
+        defaultLocale: 'en',
+        locales: ['en', 'fr'],
+      })),
+  }));
+});
+
+afterEach(() => {
+  vi.clearAllMocks();
+});
+
+describe('useScopedI18n', () => {
+  it('should log error if locale not set in next.config.js', () => {
+    const spy = vi.spyOn(console, 'error').mockImplementation(() => null);
+    const { useScopedI18n, I18nProvider } = createI18n<typeof import('./utils/en').default>({
+      en: () => import('./utils/en'),
+      fr: () => import('./utils/fr'),
+    });
+
+    function App() {
+      const t = useScopedI18n('namespace.subnamespace');
+
+      return <p>{t('hello')}</p>;
+    }
+
+    render(
+      <I18nProvider locale={en}>
+        <App />
+      </I18nProvider>,
+    );
+    expect(console.error).toHaveBeenCalledWith(
+      "[next-international] 'i18n.defaultLocale' not defined in 'next.config.js'",
+    );
+
+    spy.mockReset();
+  });
+
+  it('should log error if locales not set in next.config.js', () => {
+    const spy = vi.spyOn(console, 'error').mockImplementation(() => null);
+    const { useScopedI18n, I18nProvider } = createI18n<typeof import('./utils/en').default>({
+      en: () => import('./utils/en'),
+      fr: () => import('./utils/fr'),
+    });
+
+    function App() {
+      const t = useScopedI18n('namespace.subnamespace');
+
+      return <p>{t('hello')}</p>;
+    }
+
+    render(
+      <I18nProvider locale={en}>
+        <App />
+      </I18nProvider>,
+    );
+    expect(console.error).toHaveBeenCalledWith("[next-international] 'i18n.locales' not defined in 'next.config.js'");
+
+    spy.mockReset();
+  });
+
+  it('should throw if not used inside I18nProvider', () => {
+    const { useScopedI18n } = createI18n<typeof import('./utils/en').default>({
+      en: () => import('./utils/en'),
+      fr: () => import('./utils/fr'),
+    });
+
+    function App() {
+      const t = useScopedI18n('namespace.subnamespace');
+
+      return <p>{t('hello')}</p>;
+    }
+
+    expect(() => render(<App />)).toThrowError('`useI18n` must be used inside `I18nProvider`');
+  });
+
+  it('should translate', async () => {
+    const { useI18n, I18nProvider } = createI18n<typeof import('./utils/en').default>({
+      en: () => import('./utils/en'),
+      fr: () => import('./utils/fr'),
+    });
+
+    function App() {
+      const t = useI18n();
+
+      return <p>{t('hello')}</p>;
+    }
+
+    render(
+      <I18nProvider locale={en}>
+        <App />
+      </I18nProvider>,
+    );
+
+    expect(screen.getByText('Hello')).toBeInTheDocument();
+  });
```

Comment:
We can remove this test which uses `useI18n`, but keep the next one which uses `useScopedI18n`